New Getting Started (#152)
bump-sh/docs: src/_help/getting-started/concepts.md

---

> title: Main concepts

Reviewer: Don't we want to have the same title than in the index, which is "Important Concepts"?

We provide a detailed introduction to API definitions [through this guide](https://docs.bump.sh/guides/api-basics/api-contracts-extended-introduction/).

## API contract

An API contract represents the use of an API definition when it defines an agreement, a contract between API developers, its consumers, how it is used, and the tools that surround it.

## API document

An API document refers to the file containing an API definition.<br>

## Specification

Specifications are the official standards that define a format for describing an API, such as OpenAPI or AsyncAPI. These standards outline a set of elements and rules to follow when writing an API definition.

To learn more about the supported specifications, we have written dedicated guides on [OpenAPI](https://docs.bump.sh/guides/openapi/what-is-openapi/) and [AsyncAPI](https://docs.bump.sh/guides/asyncapi/what-is-asyncapi/).

## Deployment

A deployment is the processing of an API document by Bump.sh: after the upload, it validates the definition against its specification, analyses its content and prepares the [release](/help/publish-documentation/deploy-and-release-management/).

## Release

> The step following a successful deploy. During the [release](/help/publish-documentation/deploy-and-release-management/), the documentation is updated, the changelog reflects the related changes, and notifications, if configured, are sent out. Release step can be automatic, or manual.

Reviewer: ```suggestion
The step following a successful deploy. During the [release](/help/publish-documentation/deploy-and-release-management/), the documentation is updated, the changelog reflects the related changes, and notifications, if configured, are sent out. The release step can be automatic or manual.
```

---
title: Main concepts
---

- TOC
{:toc}

Before we proceed further, we wanted to share some vocabulary to clarify what we are talking about throughout this documentation.<br>
The terminology used here aims to be as accurate as possible, but you may encounter different terms elsewhere. We have chosen to use the most common and meaningful terms.

## API definition

> An API definition is the representation of an API written in compliance with an existing specification format.<br>

Reviewer 2: ```suggestion
An API definition is the representation of an API written in compliance with an existing specification.
```

I don't think the `format` part is useful here.

> It is sometimes referred to as an API schema, API contract, or even API specification.

Reviewer 2: We should probably say here that referring to this as an `API specification` is a strong mistake (and point to the API specification concept).

> As of today, Bump.sh supports the following specifications: Swagger, [OpenAPI](https://spec.openapis.org/oas/v3.1.0), and [AsyncAPI](https://www.asyncapi.com/docs).<br>

Reviewer 2: ```suggestion
As of today, Bump.sh supports the following specifications:[OpenAPI (Swagger)](https://spec.openapis.org/oas/v3.1.0), and [AsyncAPI](https://www.asyncapi.com/docs).<br>
```

Links are a bit inconsistent (one points on a spec version, the other on an index)

> Each standard has its prerequisites, and while your API definition doesn't need to be 100% compliant with them to be used on Bump.sh, it should generally follow their structure.

Reviewer 2: I don't think we should talk about this here: I prefer not to mention that we are cool bros allowing people to send their dirty documents, because we prefer compliant documents!

WDYT of removing this line?

Author: That's right, we should push people to keep their definition as compliant as possible.

We provide a detailed introduction to API definitions [through this guide](https://docs.bump.sh/guides/api-basics/api-contracts-extended-introduction/).

## API contract

> An API contract represents the use of an API definition when it defines an agreement, a contract between API developers, its consumers, how it is used, and the tools that surround it.

Reviewer 2: You're probably right, it doesn't deserve a paragraph here: I don't think we use it inside our help pages.

But now that @jayk-u is an expert on these concepts, maybe he could write a blog post on these terms?

> An API contract represents the use of an API definition when it defines an agreement, a contract between API developers, its consumers, how it is used, and the tools that surround it.

Author: Well... I've added this part because it was a [direct request](https://itducks.slack.com/archives/C01DP38BRTL/p1701337516321569?thread_ts=1701303858.517929&cid=C01DP38BRTL) and I thought it should be its very own entry. But I can definitely change this :p

## API document

> An API document refers to the file containing an API definition.<br>
> Bump.sh supports YAML and JSON formats for API documents.

Reviewer 2: ```suggestion
An API document refers to the file containing an API definition.
```

As we'll soon add WSDL (XML) and GraphQL schemas, we don't want to start listing the formats here, imo.

## Specification

Specifications are the standards that define a format for describing an API, such as OpenAPI or AsyncAPI. These standards outline a set of elements and rules to follow when writing an API definition.

To learn more about the supported specifications, we have written dedicated guides on [OpenAPI](https://docs.bump.sh/guides/openapi/what-is-openapi/) and [AsyncAPI](https://docs.bump.sh/guides/asyncapi/what-is-asyncapi/).

## Deploy

> During the upload of an API document, Bump.sh begins processing it to prepare for publication. During this step, various checks are performed, including ensuring that the file is functional.<br>

Reviewer 2: ```suggestion
A deploy is the processing of an API document by Bump.sh: after the upload, it validates the definition against its specification, analyses its content and prepares the [release](add/a/link/to/release).
```

Btw I have a doubt between `deploy` and `deployment`. @paulRbr do you have an opinion here?

Author: Temporarily moved on with "deployment" (deploy is a verb, would like confirmation of its usage in tech field from you guys)

> This step is referred to as deployment.

Reviewer 2: ```suggestion
```

## Release

> Once the deployment is completed, the release allows the publication of its results: the documentation is updated, the changelog reflects the changes, and notifications, if configured, are sent out.

Reviewer 2: ```suggestion
It the the step following a successful deploy. During the release, the documentation is updated, the changelog reflects the related changes, and notifications, if configured, are sent out. Release step can be automatic, or manual.
```

> ## API change
> 
> An API change refers to a change in the structure of an API (rather than a change in documentation) after the deployment of an API definition. It helps identify modifications that may impact API consumers.

Reviewer 2: ```suggestion
## API change

An API change refers to a change in the structure of an API. After the release of an API definition, it is represented by a new entry in the [API changelog](add/link).
```
